File: BOJ/18115/A.cpp
#include <iostream>
#include <deque>
using namespace std;

int main()
{
    freopen("input.txt", "r", stdin);
    ios::sync_with_stdio(false);
    cin.tie(NULL);
    int N = 0;
    cin >> N;
    deque<int> arr;
    deque<int> used;
    deque<int> ans;
    for (int i = 0; i < N; i++)
    {
        int temp;
        cin >> temp;
        used.push_back(temp);
        arr.push_back(i + 1);
    }
    for (int i = N - 1; i > -1; i--)
    {
        if (used[i] == 1)
        {
            ans.push_front(arr[0]);
            arr.erase(arr.begin());
        }
        else if (used[i] == 2)
        {
            ans.insert(ans.begin() + 1, arr[0]);
            arr.erase(arr.begin());
        }
        else if (used[i] == 3)
        {
            ans.push_back(arr[0]);
            arr.erase(arr.begin());
        }
    }
    for (int i = 0; i < ans.size(); i++)
    {
        cout << ans[i] << " ";
    }
}

File: BOJ/18115/A.py
import sys

sys.stdin = open('D:\Projects\AA\BOJ\\18115\input.txt')
input = sys.stdin.readline

N = int(input().rstrip())

used = list(map(int, input().rstrip().split()))

arr = [i for i in range(1, N+1)]
result = []

for i in range(N-1, -1, -1):
    if used[i] == 1:
        result.insert(0, arr[0])
        arr.remove(arr[0])
    elif used[i] == 2:
        result.insert(1, arr[0])
        arr.remove(arr[0])
    elif used[i] == 3:
        result.append(arr[0])
        arr.remove(arr[0])

print(result)
